metaseq/data/iterators.py

```diff
@@ -121,24 +121,28 @@ def __init__(self, iterable):
                 "please install with: pip install more_itertools"
             )
         self._peekable_itr = more_itertools.peekable(iterable)
-        self._countable_itr = more_itertools.countable(self._peekable_itr)
+
+        self.num_workers = 1 if num_workers == 0 else num_workers
+        self.batch_size = batch_size
+        self.num_shards = num_shards
+
+        self.n = 0
+        self.sentences_consumed = [0 for _ in range(self.num_workers)]
 
     def __iter__(self):
         return self
 
     def __next__(self):
-        return next(self._countable_itr)
+        self.sentences_consumed[self.n % self.num_workers] += self.batch_size * self.num_shards
+        self.n += 1
+        return next(self._peekable_itr)
 
     def __len__(self):
         return 0
 
     def has_next(self):
         return bool(self._peekable_itr)  # whether peekable has items
 
-    @property
-    def n(self):
-        return self._countable_itr.items_seen
```

Comment:
It is still there, but it is a attribute of the object now. I had to remove the `_countable_itr` because I needed to keep track of the per-worker amount read as well. Doing it as a per-worker count means that we can change the BATCH_SIZE on reload and still restore to the same spot (otherwise, we would have to know all the previous batch sizes to figure out how iteration maps to amount of data we have read through already).